hash_practice/exercises.py

```diff
@@ -2,18 +2,47 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: 0(n)
+        Space Complexity: O(n)
     """
-    pass
+    if not strings:
+        return []
+
+    map={}
+    for string in strings:
+        sorted_string = ''.join(sorted(string))
+        if sorted_string in map:
+            map[sorted_string].append(string)
+        else:
+            map[sorted_string]=[string]
+
+    result=[]
+    for value in map.values():
+        result.append(value)
+    return result
+
 
-def top_k_frequent_elements(nums, k):
+def top_k_frequent_elements(nums,k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """

```

Comment:
👍 Looking at the loop from lines 40-43, the loop runs k times and each time finds the max value of a collection roughly n long.  The time complexity is O(nk).